system_zero/src/column_layout.rs

```diff
@@ -0,0 +1,88 @@
+//// CORE REGISTERS
+
+use plonky2::hash::hashing::SPONGE_WIDTH;
+use plonky2::hash::poseidon;
+
+//// CORE REGISTERS
+
+/// A cycle counter. Starts at 0; increments by 1.
+pub(crate) const COL_CLOCK: usize = 0;
```

Comment:
Hm, I feel like these consts may be okay without a "namespace", since the crate they're in is just for System Zero, and they're crate-private. This is only a weak preference though, as the only downside is slightly greater code size from the `SystemZero::`s.

Looking back, I did put some methods like `generate_permutation_unit` under `SystemZero`, even though it doesn't use `SystemZero` objects or methods. I didn't have a very good reason for that, except that it lets us declare generics once for multiple methods.

Let me know if you have thoughts about where to draw the line, in terms of associated items vs global items.

Reply:
I definitely don't have a strong opinion on this. I think namespacing these constants/functions could help with autocomplete and may be more in line with the rest of the codebase which doesn't really use unscoped constants.
But I agree that having them in the System Zero crate is enough to avoid confusion.
So as you want, it doesn't change much in the end.